.github/workflows/test.yml

```diff
@@ -788,6 +788,89 @@ jobs:
         with:
           commit_message: Benchmark result for ${{ steps.date.outputs.pretty }} (${{ github.sha }})
 
+  next_js_integration:
+    needs: [determine_jobs, rust_prepare, rust_build_release]
+    if: needs.determine_jobs.outputs.rust == 'true'
+    name: Next.js integration test
+    runs-on: ubuntu-latest-8-core-oss
+    strategy:
+      fail-fast: false
+      matrix:
+        node: [16, 18]
+        group: [1, 2]
+    env:
+      # Enabling backtract will makes snapshot tests fail
+      RUST_BACKTRACE: 0
+      NEXT_TELEMETRY_DISABLED: 1
+      NEXT_TEST_JOB: 1
+      __INTERNAL_CUSTOM_TURBOPACK_BINARY: ${{ github.workspace }}/artifacts/release-x86_64-unknown-linux-musl/next-dev
+
+    steps:
+      # https://github.com/actions/virtual-environments/issues/1187
+      - name: tune linux network
+        run: sudo ethtool -K eth0 tx off rx off
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          repository: vercel/next.js
+          ref: v13.0.5-canary.5
+
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2.2.4
+        with:
+          version: 7.3.0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: pnpm
+
+      - name: Download binary
+        uses: actions/download-artifact@v3
+        with:
+          path: artifacts
+
+      - name: Validate next-dev binary
+        run: |
+          chmod +x ${{ github.workspace }}/artifacts/release-x86_64-unknown-linux-musl/next-dev
+          ${{ github.workspace }}/artifacts/release-x86_64-unknown-linux-musl/next-dev --display-version
+
+      - name: Install dependencies
+        run: |
+          corepack disable
+          pnpm install
+          pnpm run build
+
+      - run: npm i -g playwright-chromium@1.22.2 && npx playwright install-deps
+        timeout-minutes: 15
+
+      - run: npx @replayio/playwright install chromium
+
+      - run: |
+          chmod +x ${{ github.workspace }}/artifacts/release-x86_64-unknown-linux-musl/next-dev
```

Comment:
Do we need this line in every step?
Also, can you move the script to the bottom below `name` and `env`? 🙃

Reply:
This (chmod +x) is coming from digging unexpected test hanging issue, left it for the precaution as this is minor and easy to take out once things are stablized.

I just realized the order for name / env - this is what current next.js CI steps are written and I didn't recognize somehow. 🤔